corehq/apps/app_manager/tests/test_suite.py

```diff
@@ -317,6 +317,67 @@ def test_case_tile_suite(self):
         self._test_generic_suite("app_case_tiles", "suite-case-tiles")
 
 
+class ModuleAsChildTest(SuiteTest):
+
+    def __init__(self, args, **kwargs):
+        self.app = Application.new_app('domain', "Untitled Application", application_version=APP_V2)
+        self.module_0 = self.app.add_module(AdvancedModule.new_module('parent', None))
+        self.module_0.unique_id = 'm1'
+        self.module_1 = self.app.add_module(AdvancedModule.new_module("Untitled Module", None))
+        self.module_1.unique_id = 'm2'
+
+        for m_id in range(2):
+            self.app.new_form(m_id, "Form", None)
+        super(ModuleAsChildTest, self).__init__(args, **kwargs)
+
+    def test_advanced_module_after_test1(self):
```

Comment:
hey @snopoke can you take a look at the tests and tell me if I am going on the right path? There are refactors like saving `XML` to a file, renaming tests etc. But with this I am just trying to get the ways tests should work right.